lib/internal/async_context_frame.js

```diff
@@ -23,6 +26,7 @@ class ActiveAsyncContextFrame extends SafeMap {
 
   static set(frame) {
     setContinuationPreservedEmbedderData(frame);
+    onSet.publish(frame);
```

Comment:
Should we expose/document the `AsyncContextFrame` class via public API?
With doing just this users get some undocumented, internal thing into their hand and all they can do is to guess what it is, which API it exposes.
Any internal change on `AsyncContextFrame` would be breaking for all users of this channel.

Reply:
I discussed this with @Qard; he thinks (but can of course speak for himself here 😁) that since all channels are by default experimental, we aren't promising stability in what's returned here. As far as I'm concerned, I would've been fine with not even exposing the parameter at all.

Reply:
One of the probles to never stabilize async_hooks was because it leaks internal.
We should avoid to add again experimental APIs leaking internal as they can't be stabilized.

If the frame is not needed it's likely best to remove it. Alternative could be to replace it by a well documented wrapper object exposing only parts of AsyncContextFrame or other data like the cause for the set,...